scripts/chat.py

```diff
@@ -53,72 +53,71 @@ def chat_with_ai(
         full_message_history,
         permanent_memory,
         token_limit):
-    """Interact with the OpenAI API, sending the prompt, user input, message history, and permanent memory."""
+    """
+    Interact with the OpenAI API, sending the prompt, user input, message history, and permanent memory.
+
+    Args:
+    prompt (str): The prompt explaining the rules to the AI.
+    user_input (str): The input from the user.
+    full_message_history (list): The list of all messages sent between the user and the AI.
+    permanent_memory (Obj): The memory object containing the permanent memory.
+    token_limit (int): The maximum number of tokens allowed in the API call.
+
+    Returns:
+    str: The AI's response.
+    """
     while True:
-        try:
```

Comment:
This looks like a lot of changes. Its just indentation from moving the try block